### PR TITLE
[ENHANCEMENT] Use TLS config in oauth client config

### PR DESCRIPTION
**Pull request by Nexucis**
use TLS config when configuring the oauth client in `pkg/client`

### CODE REVIEW

**Comment by Nexucis** [comment]
part of #2311

File: pkg/client/config/config.go [hunk 1]
```diff
@@ -164,19 +159,32 @@ func NewRESTClient(config RestConfigClient) (*perseshttp.RESTClient, error) {
...
 		}
+		if config.Auth.Oauth != nil {
+			oauthConfig := &clientcredentials.Config{
+				ClientID:     config.Auth.Oauth.ClientID,
+				ClientSecret: config.Auth.Oauth.ClientSecret,
+				TokenURL:     config.Auth.Oauth.TokenURL,
+				Scopes:       config.Auth.Oauth.Scopes,
+				AuthStyle:    config.Auth.Oauth.AuthStyle,
+			}
+			ctx := context.WithValue(context.Background(), oauth2.HTTPClient, roundTripper)
```
Comment (celian-garcia) [inline pkg/client/config/config.go:178]: I think this the http client that should go under this context value, not the transport

Comment (Nexucis) [reply pkg/client/config/config.go:178]: oh yeah good call, thanks !

Comment (Nexucis) [reply pkg/client/config/config.go]: done

File: pkg/client/config/config.go [hunk 2]
```diff
@@ -119,6 +120,18 @@ type Auth struct {
 	BasicAuth  *secret.BasicAuth `json:"basic_auth,omitempty" yaml:"basic_auth,omitempty"`
 }
 
+func (a *Auth) Validate() error {
+	if a == nil {
+		return nil
+	}
+	if (a.NativeAuth != nil && (a.Oauth != nil || a.BasicAuth != nil)) ||
```
Comment (AntoineThebaud) [inline pkg/client/config/config.go:127]: I'd replace

```go
if (a.NativeAuth != nil && (a.Oauth != nil || a.BasicAuth != nil)) ||
	(a.Oauth != nil && (a.NativeAuth != nil || a.BasicAuth != nil)) ||
	(a.BasicAuth != nil && (a.NativeAuth != nil || a.Oauth != nil)) {
	return fmt.Errorf("only one type of authentication should be configured")
}
```

by

```go
nbAuthConfigured := 0
if a.NativeAuth != nil {
	nbAuthConfigured++
}
if a.Oauth != nil {
	nbAuthConfigured++
}
if a.BasicAuth != nil {
	nbAuthConfigured++
}

if nbAuthConfigured > 1 {
	return fmt.Errorf("only one type of authentication should be configured")
}
```

**Review by celian-garcia — APPROVED** [review]

**Review by AntoineThebaud — APPROVED** [review]

File: pkg/client/config/config.go (hunk 2 above)
Comment (Nexucis) [reply pkg/client/config/config.go:127]: oh yeah I like that. Thanks !

**Pull request merged**